routes/model/model.js

```diff
@@ -1,12 +1,15 @@
 const
     express = require('express'),
-    authenticate_token = require('../middleware/authenticate_token');
+    authenticate_token = require('../middleware/authenticate_token'),
+    authenticate_admin = require('../middleware/authenticate_admin');
 
 const
     config = require('../../config'),
     knex = require('knex')(config),
     RESTAPI_TYPES = require('../enum/restapi_types');
 
+const REQUIRE_ADMIN = ['employees', 'products'];
```

Comment:
This solution works well, but does not follow the Dependency Inversion Principle of SOLID.

“Abstractions should not depend on details. Details should depend on abstractions.”

Instead we should add a constructor parameter to RouteModel which enables the authenticate_admin middleware. This parameter should be an array of RESTAPI_TYPE items.